key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala

```diff
@@ -106,47 +155,93 @@ class WalletStorageTest extends BitcoinSWalletTest with BeforeAndAfterEach {
       val seedPath = getSeedPath(walletConf)
       Files.write(seedPath, badJson.getBytes())
 
-      assertThrows[InvalidData] {
-        WalletStorage.decryptMnemonicFromDisk(seedPath,
-                                              BIP39KeyManager.badPassphrase)
+      val read =
+        WalletStorage.decryptMnemonicFromDisk(seedPath, passphrase)
+
+      read match {
+        case Left(JsonParsingError(_))  => succeed
+        case res @ (Left(_) | Right(_)) => fail(res.toString)
       }
   }
 
-  it must "fail to read a mnemonic with bad password" in { walletConf =>
-    val _ = getAndWriteMnemonic(walletConf)
-    val seedPath = getSeedPath(walletConf)
-    val read = WalletStorage.decryptMnemonicFromDisk(seedPath, badPassphrase)
+  it must "fail to read an unencrypted mnemonic with improperly formatted creation time" in {
+    walletConf =>
+      val badJson =
+        """
+          | {
+          |   "mnemonicSeed":["stage","boring","net","gather","radar","radio","arrest","eye","ask","risk","girl","country"],
+          |   "creationTime":"not a number"
+          | }
+    """.stripMargin
+      val seedPath = getSeedPath(walletConf)
+      Files.write(seedPath, badJson.getBytes())
 
-    read match {
-      case Right(mnemonic) =>
-        fail("Wrote and read with different passwords")
-      case Left(DecryptionError) => succeed
-      case Left(err)             => fail(err.toString)
-    }
+      val read =
+        WalletStorage.decryptMnemonicFromDisk(seedPath, None)
+
+      read match {
+        case Left(JsonParsingError(_))  => succeed
+        case res @ (Left(_) | Right(_)) => fail(res.toString)
+      }
   }
 
-  it must "fail to read a mnemonic that has bad JSON in it" in { walletConf =>
-    val badJson =
-      """
-        | {
-        |   "iv":"ba7722683dad8067df8d069ee04530cc",
-        |   "cipherText":,
-        |   "salt":"2b7e7d718139518070a87fbbda03ea33cdcda83b555020e9344774e6e7d08af2"
-        | }
+  it must "fail to read an encrypted mnemonic with bad password" in {
```

Comment:
nit: Can you be more specific with what passphrase is bad? I.e. `AesPassword` or bip39 password